### PR TITLE
Add dashboard component for displaying statistics and associated tests

### PR DESCRIPTION
**Pull request by edlsto**
Create dashboard pack file

Add javascript_pack_tag and div to render

Add dashboard component

Create dashboard pack file

Add javascript_pack_tag and div to render

Fix linter error

Style user dashboard

Remove CL

Add dashboard test

Add dashboard snapshot

Add styling for dashboard

Add more structure for dashboards

Adjust snapshot

Remove comment

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This adds a dashboard for displaying relevant user stats within the user profile page. It also includes tests to test the rendering of the dashboard component.

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/javascript/dashboard/__tests__/__snapshots__/dashboard.test.jsx.snap [hunk 1]
```diff
@@ -0,0 +1,180 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
```
Comment (iEv0lv3) [inline app/javascript/dashboard/__tests__/__snapshots__/dashboard.test.jsx.snap:2]: Great test coverage here! I have a question for my own understanding: is Jest testing that the HTML elements exist and render within the browser?

File: app/javascript/dashboard/dashboard.jsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        </div>
+        <div className="comments stats">
+          <div className="title">COMMENTS</div>
+          <div className="stat-row">
+            <img className="icons" src={displayComment}></img>
+            <div className="stat-display">
+              <div className="display-big">{comments}</div> total
+            </div>
+          </div>
+        </div>
```
Comment (iEv0lv3) [inline app/javascript/dashboard/dashboard.jsx:48]: Excellent HTML syntax structure, it's very easy to read and understand.

File: app/javascript/packs/dashboard.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      root.firstElementChild,
+    );
+  }
+}
+
+window.InstantClick.on('change', () => {
+  loadElement();
+});
+
+loadElement();
```
Comment (iEv0lv3) [inline app/javascript/packs/dashboard.jsx:29]: This is very cool! Is this the Preact component that renders the data within the view?

**Review by iEv0lv3 — APPROVED** [review]
This PR looks excellent; thank you for all the hard work! The overall UI/UX appearance for the statistics is very professional, clean, and modern. Including the icons was a nice touch and highlights the individual statistics very well. I love the responsive design, and it works really well; from my BE perspective it seems like magic. I don't have experience testing in JavaScript, but I was able to identify the different components that make up the statistics section which indicates the coverage is complete 👍 

I'll see if I can create a few nice seeds (fake data) tonight or tomorrow so that we can display some data other than 0 for our presentation and demo.

**Comment by iEv0lv3** [comment]
Nice rebase workflow implemented on this branch! Very clean.

File: app/javascript/dashboard/__tests__/__snapshots__/dashboard.test.jsx.snap (hunk 1 above)
Comment (edlsto) [reply app/javascript/dashboard/__tests__/__snapshots__/dashboard.test.jsx.snap:2]: Yes, that is correct. It is also ensuring that the data that is passed into the component as props renders correctly.

File: app/javascript/dashboard/dashboard.jsx (hunk 2 above)
Comment (edlsto) [reply app/javascript/dashboard/dashboard.jsx:48]: Thank you!

File: app/javascript/packs/dashboard.jsx (hunk 3 above)
Comment (edlsto) [reply app/javascript/packs/dashboard.jsx:29]: Yes, that is correct.

**Pull request merged**